File: solutions/elixir/anagram/1/lib/anagram.ex
defmodule Anagram do
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.
  """
  def frequencies(str),
    do: str |> String.downcase() |> String.graphemes() |> Enum.frequencies()

  @spec match(String.t(), [String.t()]) :: [String.t()]
  def match(base, candidates) do
    freq_base = frequencies(base)

    Enum.filter(candidates, fn candidate ->
      String.downcase(candidate) != String.downcase(base) and
        frequencies(candidate) == freq_base
    end)
  end
end

File: solutions/clojure/anagram/1/src/anagram.clj
(ns anagram
  (:require [clojure.string :as s]))

(def lowercase-freqs (comp frequencies s/lower-case))

(defn anagrams-for [word prospect-list]
  (let [word-freqs (lowercase-freqs word)
        not-self (filter #(not= (s/lower-case %)
                                (s/lower-case word))
                         prospect-list)]
    (filter #(= (lowercase-freqs %) word-freqs)
            not-self)))